src/runtime/PythonTypes/PyObject.cs

```diff
@@ -136,14 +136,14 @@ public IntPtr Handle
         /// Given an arbitrary managed object, return a Python instance that
         /// reflects the managed object.
         /// </remarks>
-        public static PyObject FromManagedObject(object ob)
+        public static PyObject FromManagedObject(object ob, Type? type = null)
         {
```

Comment:
We need a use case and a test for this change as well.

Also, you have to add a separate overload because AFAIK adding default arguments breaks binary compatibility (e.g. libs compiled prior to this change will fail with `MissingMethodException` when they'll try to call the original method)